library/network/src/modules/SuSEFirewall2Services.rb

```diff
@@ -0,0 +1,608 @@
+      if !ServiceDefinedByPackage(service)
+        log.error "Service #{service} is not defined by package"
+        return nil
+      end
+
+      service[/\A#{DEFINED_BY_PKG_PREFIX}(.*)/, 1]
+    end
+
+    # Returns SCR Agent definition.
+    #
+    # @return [Yast::Term] with agent definition
+    # @param filefullpath [String] full filename path (to read by this agent)
+    # @api private
+    def GetMetadataAgent(filefullpath)
+      term(
+        :IniAgent,
+        filefullpath,
+
+        "options"  => [
+          "global_values",
+          "flat",
+          "read_only",
+          "ignore_case_regexps"
+        ],
+        "comments" => [
+          # jail followed by anything but jail (immediately)
+          "^[ \t]*#[^#].*$",
+          # comments that are not commented key:value pairs (see "params")
+          # they always use two jails
+          "^[ \t]*##[ \t]*[^([a-zA-Z0-9_]+:.*)]$",
+          # comments with three jails and more
+          "^[ \t]*###.*$",
+          # jail alone
+          "^[ \t]*#[ \t]*$",
+          # (empty space)
+          "^[ \t]*$",
+          # sysconfig entries
+          "^[ \t]*[a-zA-Z0-9_]+.*"
+        ],
+        "params"   => [
+          # commented key:value pairs
+          # e.g.: ## Name: service name
+          { "match" => ["^##[ \t]*([a-zA-Z0-9_]+):[ \t]*(.*)[ \t]*$", "%s: %s"] }
+        ]
+
+      )
+    end
+
+    # Returns service definition.
+    # See @services for the format.
+    # If *silent* is `false` (the default), the method throws an exception
+    # {Yast::SuSEFirewalServiceNotFound} if service is not found on disk.
+    #
+    # @param [String] service name (including the "service:" prefix)
+    # @param [String] (optional) whether to silently return nil
+    #                 when service is not found (default false)
+    # @api private
+    def service_details(service_name, silent = false)
+      service = all_services[service_name]
+      if service.nil? && !silent
+        log.error "Unknown service '#{service_name}'"
+        log.info "Known services: #{all_services.keys}"
+
+        raise(
+          SuSEFirewalServiceNotFound,
+          _("Service with name '%{service_name}' does not exist") % { service_name: service_name }
+        )
+      end
+
+      service
+    end
+
+    # Reads definition of services that can be used in FW_CONFIGURATIONS_[EXT|INT|DMZ]
+    # in SuSEfirewall2.
+    #
+    # @return [Boolean] if successful
+    # @api private
+    def ReadServicesDefinedByRPMPackages
+      log.info "Reading SuSEfirewall2 services from #{SERVICES_DIR}"
+      @services ||= {}
+
+      if !FileUtils.Exists(SERVICES_DIR) ||
+          !FileUtils.IsDirectory(SERVICES_DIR)
+        log.error "Cannot read #{SERVICES_DIR}"
+        return false
+      end
+
+      all_definitions = SCR.Read(path(".target.dir"), SERVICES_DIR)
+      all_definitions.reject! do |service|
+        IGNORED_SERVICES.include?(service)
+      end
```

Comment:
in ruby you can use operator `-` see http://ruby-doc.org/core-2.2.0/Array.html#method-i-2D
so it then look like `all_definitions -= IGNORED_SERVICES`